ecs-cli/modules/cli/local/converter/converter.go

```diff
@@ -258,14 +267,42 @@ func convertToPorts(portMappings []*ecs.PortMapping) []composeV3.ServicePortConf
 	return out
 }
 
-func convertToVolumes(mountPoints []*ecs.MountPoint) []composeV3.ServiceVolumeConfig {
+// namedVolumesMap maps a named top-level volume to the host source path. Used
+// to resolve container mount points to correct host for creating bind mounts
+func namedVolumesMap(volumes []*ecs.Volume) map[string]string {
+	hostPaths := make(map[string]string)
+	for _, vol := range volumes {
+		name := aws.StringValue(vol.Name)
+		host := vol.Host
+		// NOTE: Host *shouldn't* ever be nil, as ECS should return an
+		// empty host as an empty object {}. In this case, the
+		// sourcePath will be an empty string.
+		if host != nil {
+			sourcePath := aws.StringValue(host.SourcePath)
+			hostPaths[name] = sourcePath
+		}
+	}
+
+	return hostPaths
+}
+
+// Resolves any named volumes to a bind mount source path
+func convertToVolumes(mountPoints []*ecs.MountPoint, volumes []*ecs.Volume) []composeV3.ServiceVolumeConfig {
 	out := []composeV3.ServiceVolumeConfig{}
+	mapping := namedVolumesMap(volumes)
 
 	for _, mountPoint := range mountPoints {
+		volumeName := aws.StringValue(mountPoint.SourceVolume)
+		if sourcePath, ok := mapping[volumeName]; ok {
```

Comment:
I think it's safe for us to just do:
```go
sourcePath := mapping[volumeName]
```
If volumeName is not in there the default value will be `""`, which sounds like it results in the Docker daemon giving you some path:
> If this parameter is empty, then the Docker daemon has assigned a host path for you. 

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#volumes